lib/entice/logic/MapInstance.ex

```diff
@@ -0,0 +1,52 @@
+defmodule Entice.Logic.MapInstance do
+  alias Entice.Entity
+  alias Entice.Logic.MapInstance
+
+
+  defstruct(players: %{},
+    npcs: %{},
+    map: nil)
+
+  def start_instance(map, player_entities \\ [], npc_info \\ []) do
+    {:ok, id, _pid} = Entity.start()
+    npc_entities = for _npc = %{name: name, model: model} <- npc_info,
+    do: Npc.spawn(name, model) #TODO: Implement in Npc
+    MapInstance.register(id, map, player_entities, npc_entities)
+  end
+
+  def join(entity, player_entity) do #Player-already-in-players check to be done in web?
+    entity |> Entity.update_attribute(MapInstance, fn(attrs) ->
+      attrs
+      |> Map.update(MapInstance, :players, [player_entity], fn(players) -> [players | player_entity] end)
+    end)
+  end
+
+  def leave(entity, player_entity) do
```

Comment:
so this is gonna be difficult for us to keep up-to date.
but we can use the means we have through coordination (since we'll also need to register the player in the local coordination channel) and then listen here for `:entity_leave` and kick the entities from the list then automatically. we'll need a real behaviour for that, but lets actually do that and not use the Entity.xyz etc. apis - thoughts?